### PR TITLE
Add scaling action

### PR DESCRIPTION
**Pull request by Agwen12**

### CODE REVIEW

File: executor/src/main/java/org/k8loud/executor/action/kubernetes/HorizontalKubernetesScalingAction.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import data.ExecutionExitCode;
+import data.ExecutionRS;
+import io.fabric8.kubernetes.api.model.apps.ReplicaSet;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Map;
+
+@Slf4j
+public class HorizontalKubernetesScalingAction extends K8SAction {
```
Comment (proman3419) [inline executor/src/main/java/org/k8loud/executor/action/kubernetes/HorizontalKubernetesScalingAction.java:12]:
```
HorizontalScalingAction
```
the Kubernetes part is redundant

File: executor/src/main/java/org/k8loud/executor/action/kubernetes/HorizontalKubernetesScalingAction.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+     * replicas:
+     */
+    @Override
+    public ExecutionRS perform() {
+        String name = getParams().get("resource-name");
+        String type = getParams().get("resource-type");
+        String namespace = getParams().get("namespace");
+        int replicas = Integer.parseInt(getParams().get("replicas"));
+
+
```
Comment (proman3419) [inline executor/src/main/java/org/k8loud/executor/action/kubernetes/HorizontalKubernetesScalingAction.java:37]: Newlines

File: executor/src/main/java/org/k8loud/executor/action/kubernetes/HorizontalKubernetesScalingAction.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+
+        switch (type) {
+            case "ReplicaSet" -> client.apps().replicaSets().inNamespace(namespace).withName(name).scale(replicas);
```
Comment (proman3419) [inline executor/src/main/java/org/k8loud/executor/action/kubernetes/HorizontalKubernetesScalingAction.java:41]: Can be simplified

**Review by proman3419 — CHANGES_REQUESTED** [review]

File: executor/pom.xml [hunk 4]
```diff
@@ -95,15 +95,39 @@
             <version>3.20.6</version>
             <!-- use the same version as your Camel core version -->
         </dependency>
+<!--        <dependency>-->
+<!--            <groupId>io.kubernetes</groupId>-->
+<!--            <artifactId>client-java</artifactId>-->
+<!--            <version>18.0.0</version>-->
+<!--        </dependency>-->
```
Comment (pgorgolew) [inline executor/pom.xml:102]: delete?

File: executor/src/test/java/org/k8loud/executor/action/kubernetes/HorizontalKubernetesScalingActionTest.java [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+@EnableKubernetesMockClient(crud = true)
+class HorizontalKubernetesScalingActionTest {
+
+    @Rule
+    public KubernetesServer server = new KubernetesServer(true, true);
+    KubernetesClient client;
+
+
+    @BeforeEach
+    public void setUp() {
```
Comment (pgorgolew) [inline executor/src/test/java/org/k8loud/executor/action/kubernetes/HorizontalKubernetesScalingActionTest.java:40]: it is empty so delete it

File: executor/src/test/java/org/k8loud/executor/action/kubernetes/HorizontalKubernetesScalingActionTest.java [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+
+@EnableKubernetesMockClient(crud = true)
+class HorizontalKubernetesScalingActionTest {
+
+    @Rule
+    public KubernetesServer server = new KubernetesServer(true, true);
```
Comment (pgorgolew) [inline executor/src/test/java/org/k8loud/executor/action/kubernetes/HorizontalKubernetesScalingActionTest.java:35]: can be deleted? No usage

File: executor/src/test/java/org/k8loud/executor/action/kubernetes/HorizontalKubernetesScalingActionTest.java [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+                Map.of("resource-type", "ReplicaSet", "resource-name", "repl1", "namespace", "test", "replicas", "3"),
+                Map.of("resource-type", "ReplicaSet", "resource-name", "repl1", "namespace", "test", "replicas", "1"),
+                Map.of("resource-type", "ReplicaSet", "resource-name", "repl1", "namespace", "test", "replicas", "2")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideReplicasetScalingParams")
+    void testScalingReplicaSet(Map<String, String> params) {
+
```
Comment (pgorgolew) [inline executor/src/test/java/org/k8loud/executor/action/kubernetes/HorizontalKubernetesScalingActionTest.java:136]: delete this break

**Review by pgorgolew — CHANGES_REQUESTED** [review]

File: executor/src/main/java/org/k8loud/executor/action/kubernetes/HorizontalKubernetesScalingAction.java (hunk 3 above)
Comment (proman3419) [reply executor/src/main/java/org/k8loud/executor/action/kubernetes/HorizontalKubernetesScalingAction.java:41]: I can't :(

**Review by proman3419 — APPROVED** [review]

**Review by pgorgolew — APPROVED** [review]

**Pull request merged**